### PR TITLE
Create summaries of hosting capacity metrics from database

### PR DESCRIPTION
**Pull request by daniel-thom**
This is currently draft. @wenbowangnrel @KwamiS I'd like you to test it as-is to ensure that it is doing what you want. I still need to perform some count checks.

Kwami's SWAD script is now in a separate PR.

### CODE REVIEW

File: disco/cli/compute_pv_swad.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        ratio_sum = sum([v / max(0.001, load_per_bus[k]) for k, v in existing_pv_dict.items()])
+        wavg_pv_distance = sum(slack_dico.values()) / max(0.001, ratio_sum)
+
+    return wavg_pv_distance
+
+
+def collect_weighted_average_pv_distance(filename: Path, option):
+    reply = dss.utils.run_command(f"compile {filename}")
+    if reply != "":
+        print(f"Failed to compile OpenDSS model {filename}: {reply}", file=sys.stderr)
```
Comment (jgu2) [inline disco/cli/compute_pv_swad.py:84]: Use logging? same for `print` in followings.

File: disco/cli/compute_pv_swad.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+                            }
+                        )
+    return results
+
+
+def main():
+    if len(sys.argv) == 1:
+        print(
+            f"Usage: python {sys.argv[0]} PV_DEPLOYMENTS_PATH [size|nodal_ratio]", file=sys.stderr
+        )
```
Comment (jgu2) [inline disco/cli/compute_pv_swad.py:126]: Not sure if it's worth a refactoring, a `click.command` may be more user friendly.

File: disco/postprocess/query.mustache [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+		JOIN job ON job.task_id = task.id
+		JOIN metadata AS mt ON mt.job_id = job.id
+		WHERE task.name LIKE '%{{task_pattern}}'
+            AND feeder NOT IN bad_feeders
+            AND mt.scenario = '{{scenario}}'
+    ;
+
+-- Find the max penetration_level for each feeder.
+DROP view if exists hc_max;
+create temp view hc_max as
```
Comment (jgu2) [inline disco/postprocess/query.mustache:109]: nit: for consistency, uppercase of SQL keywords?

File: disco/postprocess/query.mustache [hunk 4]
```diff
@@ -0,0 +1,365 @@
+CREATE INDEX IF NOT EXISTS tmidx1 ON thermal_metrics(job_id, name);
+CREATE INDEX IF NOT EXISTS vmidx1 ON voltage_metrics(job_id, name);
+CREATE INDEX IF NOT EXISTS midx1 ON metadata(job_id);
+CREATE INDEX IF NOT EXISTS jobidx1 ON job(id);
```
Comment (daniel-thom) [inline disco/postprocess/query.mustache:4]: @jgu2 These sped up queries by 2-3x.

Comment (daniel-thom) [inline disco/postprocess/query.mustache:4]: @jgu2 These sped up queries by 2-3x.

**Review by KwamiS — APPROVED** [review]

**Pull request merged**